From bf00cc20997b31b7706d1fd398f11b011b594274 Mon Sep 17 00:00:00 2001
From: trungvanle <trung.le.xk@renesas.com>
Date: Tue, 23 Jan 2024 16:16:29 +0700
Subject: [PATCH] Support bayer2raw while using mali

This commit will support bayer2raw for mali (G2L) by removing
EGLImage pixmap object creating part because mali (G2L)
does not support EGL pixmap.

Signed-off-by: trungvanle <trung.le.xk@renesas.com>
---
 bayer2raw.c | 28 ----------------------------
 bayer2raw.h |  4 ----
 2 files changed, 32 deletions(-)

diff --git a/bayer2raw.c b/bayer2raw.c
index 00b09e5..c0c598b 100644
--- a/bayer2raw.c
+++ b/bayer2raw.c
@@ -332,34 +332,6 @@ void bayer2raw_free_all_inbuf(struct context_bayer2rgb *ctx)
     free(ctx->in_buf);
 }

-int bayer2raw_create_outbuf(struct context_bayer2rgb *ctx,
-                            unsigned long address)
-{
-  struct extern_buffer buffer;
-  EGLNativePixmapTypeREL  sNativePixmap;
-  NativePixmapType eglPixmap;
-
-  sNativePixmap.width    = ctx->width;
-  sNativePixmap.height   = ctx->height;
-  sNativePixmap.stride   = ctx->width;
-  sNativePixmap.usage    = 0;
-  sNativePixmap.pixelData    = (void*)address;
-  sNativePixmap.format   = EGL_NATIVE_PIXFORMAT_ARGB8888_REL;
-  eglPixmap = (NativeWindowType)&sNativePixmap;
-  buffer.pEGLImage = eglCreateImageKHR(ctx->display,
-                                      EGL_NO_CONTEXT,
-                                      EGL_NATIVE_PIXMAP_KHR,
-                                      eglPixmap, NULL);
-
-
-  if (buffer.pEGLImage == EGL_NO_IMAGE_KHR){
-    printf("%s: Can not init eglImageKHR. %x\n", __FUNCTION__, eglGetError());
-    return -1;
-  }
-
-  return import_extern_outbuf(ctx, &buffer);
-}
-
 int bayer2raw_create_outbuf_dmabuf(struct context_bayer2rgb *ctx, int dma_fd)
 {
   struct extern_buffer buffer;
diff --git a/bayer2raw.h b/bayer2raw.h
index dd11890..6858f8a 100644
--- a/bayer2raw.h
+++ b/bayer2raw.h
@@ -3,7 +3,6 @@

 #include <EGL/egl.h>
 #include <EGL/eglext.h>
-#include <EGL/eglext_REL.h>
 #include <EGL/eglplatform.h>
 #include <GLES2/gl2.h>
 #include <GLES2/gl2ext.h>
@@ -43,9 +42,6 @@ struct context_bayer2rgb
 int init_bayer_rgb_conversion(struct context_bayer2rgb *ctx);
 int init_bayer_rgb_conversion_extern_output(struct context_bayer2rgb *ctx);

-/* Give memory address to create framebuffer store RGB surface data */
-int bayer2raw_create_outbuf(struct context_bayer2rgb *ctx,
-                            unsigned long address);
 /* Give DMABUF FD to create framebuffer store RGB surface data */
 int bayer2raw_create_outbuf_dmabuf(struct context_bayer2rgb *ctx,
                                    int dma_fd);
--
2.43.0
